File: frontend/src/interfaces/clusterRepository.ts
// frontend/src/interfaces/clusterRepository.ts

import type { Cluster, CreateClusterRequest, DiagnosisRequest, DiagnosisResponse, UpdateClusterRequest } from '../models/cluster';

export interface ClusterRepository {
  diagnoseCluster(request: DiagnosisRequest): Promise<DiagnosisResponse>;
  getCluster(id: string): Promise<Cluster>;
  listClusters(): Promise<Cluster[]>;
  createCluster(request: CreateClusterRequest): Promise<Cluster>;
  updateCluster(id: string, request: UpdateClusterRequest): Promise<Cluster>;
  deleteCluster(id: string): Promise<void>;
}

File: backend/core-api/interfaces/clusterRepository.go
// backend/core-api/interfaces/clusterRepository.go

package interfaces

import "github.com/AvinashMahala/ClusterGenie/backend/core-api/models"

type ClusterRepository interface {
	CreateCluster(cluster *models.Cluster) (*models.Cluster, error)
	GetCluster(id string) (*models.Cluster, error)
	ListClusters() ([]*models.Cluster, error)
	UpdateCluster(id string, cluster *models.Cluster) (*models.Cluster, error)
	DeleteCluster(id string) error
}
